File: Count Pairs Odd Xor/Count_Pair_Odd_Xor.cpp
class Solution {
  public:
    long long countOddXorPairs(vector<int>& arr) {
        long long odd =0,even = 0;
        for(auto i: arr) {
            if(i%2 == 0) even++;
            else odd++;
        }
        return odd*even;
    }
};

File: Count Pairs Odd Xor/Count_Pair_Odd_Xor.java
class Solution {
    public long countOddXorPairs(int[] arr) {
        long odd = 0, even = 0;

        for (int i : arr) {
            if (i % 2 == 0) {
                even++;
            } else {
                odd++;
            }
        }
        return odd * even;
    }
}